libvast/include/vast/detail/algorithms.hpp

```diff
@@ -8,11 +8,26 @@
 
 #pragma once
 
+#include "vast/concepts.hpp"
+
 #include <algorithm>
 #include <vector>
 
 namespace vast::detail {
 
+template <class T>
+concept has_contains = requires(T& t, typename T::value_type& x) {
+  requires concepts::container<T>;
+  requires t.contains(x);
+};
+
+template <concepts::range T, class U>
+bool contains(const T& t, const U& x) {
+  if constexpr (has_contains<T>)
+    return t.contains(x);
+  return std::find(t.begin(), t.end(), x) != t.end();
```

Comment:
```suggestion
  else
    return std::find(t.begin(), t.end(), x) != t.end();
```

Without this else, the function fails to compile if `t.contains(x)` is valid, but `*t.begin() == x` is not, which would be the case for maps, where dereferencing the iterator would yield the key-value pairs, while contains just takes the key.